torch/utils/_python_dispatch.py

```diff
@@ -56,7 +56,7 @@ def __enter__(self):
         return self
```

Comment:
I still don't think infra mode keys should go through TorchDispatchMode parent class. But if they do, don't you need to pass the mode key when you enter?

Reply:
Right now in the PR, this works out fine because we eventually call `_push_on_torch_dispatch_stack` (the C++ binding), which `getattr`'s the `_mode_key`.